src/ens/contract/contract.ts

```diff
@@ -0,0 +1,62 @@
+import { ResolutionError, ResolutionErrorCode } from "../..";
+import EnsProvider, { FourBytes } from "../../provider/provider";
+import BaseConnection from "../../baseConnection";
+import {defaultAbiCoder as AbiCoder} from 'ethers/utils/abi-coder';
+
+/** Contract class describes the smartContract on the etherium */
+export default class Contract extends BaseConnection {
+  readonly contractInterface: [any];
+  readonly address: string;
+  readonly provider: EnsProvider;
+
+  /**
+   * @param contractInterface JSON-RPC interface of smartContract
+   * @param address Contract's address
+   */
+  constructor(contractInterface, address: string) {
+    super();
+    this.contractInterface = contractInterface;
+    this.address = address;
+    this.provider = EnsProvider.getInstance();
+  }
+
+  /**
+   * Used to fetch a Contract method
+   * @param method - method name 
+   * @param args - method args
+   * @async
+   */
+  async fetchMethod(method: string, args: string[]): Promise<any> {
+    const methodDescription = this.contractInterface.find(
+      param =>
+        param.name === method &&
+        param.inputs.length === args.length
+    );
+    if (!methodDescription)
+      throw new ResolutionError(ResolutionErrorCode.IncorrectResolverInterface, {method: this.provider.namingService});
```

Comment:
I don't see how it is possible to throw this error because we only pass `contractInterface` from our library but not querying it somewhere.

The case when we do have to throw this error is on the error response from JSON RPC: this is where we can in fact get the error because the real contract may not have the interface described in our file.